pages/pipelines/command_step.md

```diff
@@ -306,6 +306,24 @@ _Optional Attributes_
       <em>Example:</em> <code>2</code>
     </td>
   </tr>
+  <tr>
+    <td><code>signal</code></td>
+    <td>
+      The signal sent to a process, if the process did not intercept and handle the signal<br>
```

Comment:
This sentence is a bit unclear to me. Could you please unpack it a bit more?

> The signal sent to a process

Looking at [Signal handing in the agent](https://buildkite.com/docs/agent/v3#signal-handling), this value looks like it's the name of the signal? Is "a process" here also the build job process?

> if the process did not intercept and handle the signal

This was the confusing bit for me. The previous part of the sentence sounds like the process did receive the signal. This part sounds like a caveat to say that the previous part won't happen if the process intercepts and handles the signal. But, the caveat still involves the process receiving the signal. That seems like a contradiction.

I'm probably missing something simple 🤔

Reply:
It's .. complex 😅

It's the signal sent by the agent to the bootstrap, and then likely forwarded from the bootstrap to the command process, and only appears here iff the signal was not handled by one of those interior processes and so was handled by the default linux exit handler and turned into an exit code based on the signal. I think.

The most reliable usage is for SIGKILL because there cannot be a handler for that signal, and so it propagates reliably. It's also arguably the most useful signal for rules in here, because it allows differentiating graceful cancellation from cancellation which times out, for example.

I would love better word suggestions :pray:

Reply:
I've tried another iteration on the words.